src/test/java/org/folio/circulation/domain/policy/OverdueFinePolicyTest.java

```diff
@@ -0,0 +1,101 @@
+
+  private static JsonObject overdueFinePolicyJsonObject;
+
+  @Before
+  public void setUp() {
+    overdueFinePolicyJsonObject = new JsonObject()
+      .put("id", UUID.randomUUID().toString())
+      .put("name", "Overdue Fine Policy")
+      .put("overdueFine", new JsonObject()
+        .put("intervalId", "minute")
+        .put("quantity", 5.0))
+      .put("maxOverdueFine", 10.0)
+      .put("maxOverdueRecallFine", 10.0)
+      .put("gracePeriodRecall", true)
+      .put("countClosed", true);
+  }
+
+  @Test
+  public void shouldAcceptValidJsonRepresentation() {
+    OverdueFinePolicy overdueFinePolicy = OverdueFinePolicy.from(overdueFinePolicyJsonObject);
+
+    assertThat(overdueFinePolicy.getId(), is(overdueFinePolicyJsonObject.getString("id")));
+    assertThat(overdueFinePolicy.getName(), is(overdueFinePolicyJsonObject.getString("name")));
+    assertThat(overdueFinePolicy.getOverdueFineInterval(), is(OverdueFineInterval.fromValue(
+      overdueFinePolicyJsonObject.getJsonObject("overdueFine").getString("intervalId"))));
+    assertThat(overdueFinePolicy.getOverdueFine(),
+      is(overdueFinePolicyJsonObject.getJsonObject("overdueFine").getDouble("quantity")));
+    assertThat(overdueFinePolicy.getMaxOverdueFine(),
+      is(overdueFinePolicyJsonObject.getDouble("maxOverdueFine")));
+    assertThat(overdueFinePolicy.getMaxOverdueRecallFine(),
+      is(overdueFinePolicyJsonObject.getDouble("maxOverdueRecallFine")));
+    assertThat(overdueFinePolicy.getIgnoreGracePeriodForRecalls(),
+      is(overdueFinePolicyJsonObject.getBoolean("gracePeriodRecall")));
+    assertThat(overdueFinePolicy.getCountPeriodsWhenServicePointIsClosed(),
+      is(overdueFinePolicyJsonObject.getBoolean("countClosed")));
+  }
+
+  @Test
+  public void shouldAcceptOverdueFinePolicyWithMissingOverdueFineQuantity() {
+    JsonObject jsonObject = overdueFinePolicyJsonObject.copy();
+    jsonObject.getJsonObject("overdueFine").remove("quantity");
+
+    OverdueFinePolicy overdueFinePolicy = OverdueFinePolicy.from(jsonObject);
+
+    assertThat(overdueFinePolicy, notNullValue());
+  }
+
+  @Test
+  public void shouldAcceptOverdueFinePolicyWithNullOverdueFineQuantity() {
+    JsonObject jsonObject = overdueFinePolicyJsonObject.copy();
+    jsonObject.getJsonObject("overdueFine").put("quantity", (Double) null);
+
+    OverdueFinePolicy overdueFinePolicy = OverdueFinePolicy.from(jsonObject);
+
+    assertThat(overdueFinePolicy, notNullValue());
+  }
+
+  @Test
+  public void shouldAcceptOverdueFinePolicyWithMissingOverdueFineInterval() {
+    JsonObject jsonObject = overdueFinePolicyJsonObject.copy();
+    jsonObject.getJsonObject("overdueFine").remove("intervalId");
+
+    OverdueFinePolicy overdueFinePolicy = OverdueFinePolicy.from(jsonObject);
+
+    assertThat(overdueFinePolicy, notNullValue());
+  }
+
+  @Test
+  public void shouldAcceptOverdueFinePolicyWithNullOverdueFineInterval() {
+    JsonObject jsonObject = overdueFinePolicyJsonObject.copy();
```

Comment:
I think that this is the same as `shouldAcceptOverdueFinePolicyWithMissingOverdueFineInterval` above.

`json.getString("intervalId")` will return `null` in both cases when the property does not exists and when it is set to null.

What do you think?

Reply:
Yes, but we know that because we know how it's implemented inside. From the test perspective these JSONs are different, aren't they?

Reply:
Sure, let's leave them.